lib/interactive_viewer.ml

```diff
@@ -134,3 +141,27 @@ let view (patches : Patch.t list) =
     ]
 
 let start patch = Ui_loop.run ~quit ~tick_period:0.2 (view patch)
+let convert_char_to_key (c : char) : Ui.key = (`ASCII c, [])
+
+let start_test patch events =
+  let content_ui_root = Lwd.observe (view patch) in
+
+  let rec process_events (events : char list) (image : Notty.image) :
+      Notty.image =
+    match events with
+    | [] -> image
+    | event :: rest ->
+        let ui_event = convert_char_to_key event in
+
+        let content_ui = Lwd.quick_sample content_ui_root in
+        let ui_renderer = content_image_renderer content_ui in
```

Comment:
You don't need to create a new renderer each new event!

Reply:
This too